packages/webcert/src/feature/certificate/Inputs/UeUncertainDate.tsx

```diff
@@ -0,0 +1,127 @@
+import { CertificateDataElement, Dropdown, ConfigureUeUncertainDate, QuestionValidationTexts, ValueDate } from '@frontend/common'
+import { ConfigUeDropdownItem } from '@frontend/common/src/types/certificate'
+import React from 'react'
+import styled from 'styled-components/macro'
+import { updateCertificateDataElement } from '../../../store/certificate/certificateActions'
+import { useAppDispatch } from '../../../store/store'
+
+const ValidationWrapper = styled.div`
+  flex: 0 !important;
+  flex-basis: 100% !important;
+  padding-bottom: 16px;
+  margin-top: 0;
+`
+
+export interface Props {
+  hasValidationError?: boolean
+  disabled?: boolean
+  question: CertificateDataElement
+}
+
+const UeUncertainDate: React.FC<Props> = (props) => {
+  const { question, hasValidationError, disabled } = props
+  const config = question.config as ConfigureUeUncertainDate
+  const dispatch = useAppDispatch()
+  const [selectedYear, setSelectedYear] = React.useState(getYear((question.value as ValueDate).date))
+  const [selectedMonth, setSelectedMonth] = React.useState(getMonth((question.value as ValueDate).date))
+  const [disabledMonth, setDisabledMonth] = React.useState(
+    disabled || getYear((question.value as ValueDate).date) === '0000' || getYear((question.value as ValueDate).date) === ''
+  )
+
+  const years: ConfigUeDropdownItem[] = [{ id: '', label: 'Ange år' }]
+
+  if (config.unknownYear) years.push({ id: '0000', label: '0000 (ej känt)' })
```

Comment:
I'm so boring now, I know ;) But I think it's hard to read if-statements like these. They look cool, but I don't think we write if-statements like that anywhere else in the code. What do you think?

Reply:
Just old habit. hopefully more readable now